mkt/developers/tasks.py

```diff
@@ -217,6 +217,8 @@ def pngcrush_image(src, hash_field='image_hash', **kw):
                   instance through set_modified_on
     """
     log.info('[1@None] Optimizing image: %s' % src)
+    with public_storage.open(src) as srcf, open(src, 'w') as destf:
```

Comment:
If you want to write to the full path on the local filesystem, I'd use our local_storage class so that the directories are all created for you if they don't exist. But I think I'd prefer for us to write a tempfile on the local storage instead of re-using the path, it's easier to understand and will help in the long term when we do path cleanup.